compute/apiv1/smoke_test.go

```diff
@@ -0,0 +1,290 @@
+	"time"
+)
+
+var projectId = testutil.ProjID()
+var defaultZone = "us-central1-a"
+
+func TestCreateGetListInstance(t *testing.T){
+	rand.Seed(time.Now().UTC().UnixNano())
+	name := fmt.Sprintf("gotest%d", rand.Int())
+	description := "тест"
+	machineType := fmt.Sprintf(
+		"https://www.googleapis.com/compute/v1/projects/%s/zones/%s/machineTypes/n1-standard-1",
+		projectId, defaultZone)
+	ctx := context.Background()
+	c, err := NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	zonesClient, err := NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	configName := "default"
+	accessConfig := computepb.AccessConfig{
+		Name: &configName,
+	}
+	configs := []*computepb.AccessConfig {
+		&accessConfig,
+	}
+	networkInterface := computepb.NetworkInterface{
+		AccessConfigs: configs,
+	}
+	interfaces := []*computepb.NetworkInterface{
+		&networkInterface,
+	}
+	sourceImage := "projects/debian-cloud/global/images/family/debian-10"
+	initializeParams := &computepb.AttachedDiskInitializeParams{
+		SourceImage: &sourceImage,
+	}
+	diskType := computepb.AttachedDisk_PERSISTENT
+	disk := computepb.AttachedDisk{
+		AutoDelete: proto.Bool(true),
+		Boot: proto.Bool(true),
+		Type: &diskType,
+		InitializeParams: initializeParams,
+	}
+	disks := []*computepb.AttachedDisk {
+		&disk,
+	}
+	instance := &computepb.Instance{
+		Name: &name,
+		Description: &description,
+		MachineType: &machineType,
+		Disks: disks,
+		NetworkInterfaces: interfaces,
+	}
+
+	createRequest := &computepb.InsertInstanceRequest{
+		Project: projectId,
+		Zone: defaultZone,
+		InstanceResource: instance,
+	}
+
+	insert, err := c.Insert(ctx, createRequest)
+	if err != nil {
+		fmt.Println(err)
+		t.Error(err)
+	}
+
+	waitZonalRequest := &computepb.WaitZoneOperationRequest{
+		Project: projectId,
+		Zone: defaultZone,
+		Operation: insert.GetName(),
+	}
+	_, err = zonesClient.Wait(ctx, waitZonalRequest)
+	if err != nil {
+		return
+	}
+	fmt.Printf("Inserted instance named %s\n", name)
+	defer ForceDeleteInstance(name, ctx, c)
+
+	getRequest := &computepb.GetInstanceRequest{
+		Project: projectId,
+		Zone: defaultZone,
+		Instance: name,
+	}
+	get, err := c.Get(ctx, getRequest)
+	if err != nil {
+		fmt.Println(err)
+		t.Error(err)
+	}
+	assert.Equal(t, name, get.GetName())
+	assert.Equal(t, "тест", get.GetDescription())
+
+	listRequest := &computepb.ListInstancesRequest{
+		Project: projectId,
+		Zone: defaultZone,
+	}
+
+	list, err := c.List(ctx, listRequest)
+	if err != nil {
+		fmt.Println(err)
+		t.Error(err)
+	}
+	items := list.GetItems()
+	found := false
+	for _, element := range items {
+		if element.GetName() == name {
+			found = true
+		}
+	}
+	if found == false{
+		t.Error("Couldn't find the instance in list response")
+	}
+
+	deleteInstanceRequest := &computepb.DeleteInstanceRequest{
+		Project: projectId,
+		Zone: defaultZone,
+		Instance: name,
+	}
+	_, err = c.Delete(ctx, deleteInstanceRequest)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Print the error is unnecessary because `t.Error` will do the same thing but in the context of the failure.